tests/LexerTest.php

```diff
@@ -178,23 +178,97 @@ public function testBigNumbers()
         $this->assertEquals('922337203685477580700', $node->getValue());
     }
 
-    public function testStringWithEscapedDelimiter()
+    /**
+     * @dataProvider getStringWithEscapedDelimiter
+     */
+    public function testStringWithEscapedDelimiter(string $template, string $expected, array $expectedDeprecations = [])
     {
-        $tests = [
-            "{{ 'foo \' bar' }}" => 'foo \' bar',
-            '{{ "foo \" bar" }}' => 'foo " bar',
-        ];
-
         $lexer = new Lexer(new Environment(new ArrayLoader()));
-        foreach ($tests as $template => $expected) {
-            $stream = $lexer->tokenize(new Source($template, 'index'));
-            $stream->expect(Token::VAR_START_TYPE);
-            $stream->expect(Token::STRING_TYPE, $expected);
+        $stream = $lexer->tokenize(new Source($template, 'index'));
+        $stream->expect(Token::VAR_START_TYPE);
+        $stream->expect(Token::STRING_TYPE, $expected);
 
-            // add a dummy assertion here to satisfy PHPUnit, the only thing we want to test is that the code above
-            // can be executed without throwing any exceptions
-            $this->addToAssertionCount(1);
-        }
+        $this->assertSame($expectedDeprecations, $lexer->getDeprecations());
+    }
+
+    public function getStringWithEscapedDelimiter()
+    {
+        yield '{{ \'App\\\\Test\' }} => App\Test' => [
+            <<<'TWIG'
+            {{ 'App\\Test' }}
+            TWIG,
+            <<<'EOF'
+            App\Test
+            EOF,
+            [
+                'Character \ at position 5 does not require to be escaped.'
```

Comment:
This deprecation looks weird to me. Escaping a `\` should become optional, but should stay allowed IMO (like in PHP).

Reply:
Removed it.

Can we somehow notify the developer in v4 that escaping a `\` is not needed? Maybe via `E_USER_NOTICE`? Or should we just not care about that 😊

Reply:
we should not care (just like PHP allows escaping them). Especially given that escaping them is still required in some cases

Reply:
Ok clear!